addons/mail/static/src/models/user_setting/user_setting.js

```diff
@@ -0,0 +1,321 @@
+        }
+
+        toggleLayoutSettingsWindow() {
+            this.update({ isRtcLayoutSettingDialogOpen: !this.isRtcLayoutSettingDialogOpen });
+        }
+
+        /**
+         * toggles the display of the option window
+         */
+        toggleWindow() {
+            this.update({ isOpen: !this.isOpen });
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @param {function} f
+         * @param {number} delay in ms
+         * @param {any} key
+         */
+        _debounce(f, delay, key) {
+            this._timeoutIds[key] && browser.clearTimeout(this._timeoutIds[key]);
+            this._timeoutIds[key] = browser.setTimeout(() => {
+                if (!this.exists()) {
+                    return;
+                }
+                f();
+            }, delay);
+        }
+
+        /**
+         * @private
+         */
+        _loadLocalSettings() {
+            const voiceActivationThresholdString = this.env.services.local_storage.getItem(
+                "mail_user_setting_voice_threshold"
+            );
+            const audioInputDeviceId = this.env.services.local_storage.getItem(
+                "mail_user_setting_audio_input_device_id"
+            );
+            const voiceActivationThreshold = parseFloat(voiceActivationThresholdString);
+            if (voiceActivationThreshold > 0) {
+                this.update({
+                    voiceActivationThreshold,
+                    audioInputDeviceId,
+                });
+            }
+        }
+
+        /**
+         * @private
+         */
+        async _saveSettings() {
+            this._debounce(async () => {
+                await this.async(() => this.env.services.rpc(
+                    {
+                        model: 'res.users.settings',
+                        method: 'set_res_users_settings',
+                        args: [[this.messaging.currentUser.resUsersSettingsId], {
+                            push_to_talk_key: this.pushToTalkKey,
+                            use_push_to_talk: this.usePushToTalk,
+                            voice_active_duration: this.voiceActiveDuration,
+                        }],
+                    },
+                    { shadow: true },
+                ));
+            }, 2000, 'globalSettings');
+        }
+
+    }
+
+    UserSetting.fields = {
+        /**
+         * DeviceId of the audio input selected by the user
+         */
+        audioInputDeviceId: attr({
+            default: '',
+        }),
+        id: attr(),
+        /**
+         * true if the dialog for the call viewer layout is open
+         */
+        isRtcLayoutSettingDialogOpen: attr({
+            default: false,
+        }),
+        messaging: one2one('mail.messaging', {
+            inverse: 'userSetting',
+        }),
```

Comment:
`messaging` field are automatically set on all records, so don't need to define again.